internal/controller/components/suite_test.go

```diff
@@ -62,14 +62,6 @@ var _ = BeforeSuite(func() {
 	Expect(err).NotTo(HaveOccurred())
 	Expect(cfg).NotTo(BeNil())
 
-	DeferCleanup(func() {
-		By("DeferCleanup: tearing down the test environment")
-		if testEnv != nil {
-			err := testEnv.Stop()
-			Expect(err).NotTo(HaveOccurred())
-		}
-	})
-
```

Comment:
Why it is removed?

Reply:
It was removed because it was duplicating the cleanup logic already handled by the AfterSuite block

Reply:
This follows Ginkgo best practices where AfterSuite handles suite-level cleanup and DeferCleanup is reserved for individual test cleanup when needed.